api/controllers/user.go

```diff
@@ -83,15 +86,95 @@ func HandleSignUp(c *gin.Context) {
 		return
 	}
 
-	data := interfaces.User{Username: form.Username, Email: form.Email, Password: string(hashed), IsVerified: false}
+	//generate code
+	validationCode := utils.GetValidationCode()
+
+	data := interfaces.User{Username: form.Username,
+		Email:          form.Email,
+		Password:       string(hashed),
+		ValidationCode: validationCode,
+		TimeExpiration: time.Now().Add(time.Minute * 60),
+		IsVerified:     false}
 
-	//Insert user in database
 	err = models.InsertUser(data)
 
+	if err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
+		return
+	}
+
+	//send mail of verification
+	err = utils.SendEmail(form.Email, "Here is your validation code", validationCode)
 	if err != nil {
 		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
 		return
 	}
 
 	c.IndentedJSON(http.StatusOK, gin.H{"message": "User created successfully"})
 }
+
+func HandleCodeValidation(c *gin.Context) {
+	var form interfaces.ValidationCode
```

Comment:
So... the user has to send the timeExpiration in the request body? I got a Bad Request response when sent the following payload: 

```json
{
	"email": "user@gmail.com", 
	"validationCode": "012345",
}
```